### DISCUSSION TITLE
Mysterium Node

### DISCUSSION THREAD

**Discussion by AntonGreydon** [MRColorR/money4band#21] [Ideas]
Great Package you put together here. Thanks for that. 
Have you concidered adding a Mysterium Node? They pay fairly well and shouldn't be too hard to include. Might even want to concider adding it by replacing Bitping. That damn thing will not earn you a dollar in a thousand years.

**Comment by MRColorR** [comment]
> Great Package you put together here. Thanks for that. Have you concidered adding a Mysterium Node? They pay fairly well and shouldn't be too hard to include. Might even want to concider adding it by replacing Bitping. That damn thing will not earn you a dollar in a thousand years.

hello, Anton, I'm glad you like the project it means a lot to me🙂.
I always try to improve the project to add new functions and apps into it. I already use Mystnode but i decided to exclude it from this project for a couple of reasons:

Regarding Myst VPN:
1. it is a dVPN and by default it open your node and also IP and connection to every kind of users. You need to manually configure it to retrict avaiablility to only secure and cetified business contract. 
2. it require a manual multi-step configuration and you need to pay some MYST to start a new node if free nodes slots are not avaiable.
3. it require a lot of bandwidth , several mbps up to gbps to be a valuable node with good qualty (hig bitrate and uptime, low ping) to increase the chances to be choosen.

The apps in this project need to be, free, simple to setup and run, like "deploy and forget". They also need to be fair safe to use by default without need of manual restrictions and configurations. 

I'm glad you proposed adding Myst, I've been thinking about it for some time already. Given the reasons described above, if more people are interested in it, I see two ways: I could publish a separate project with a  self updating and security hardened mystnode deploy OR just add it as an optional deploy in the auto configuration script.

Regarding Bitping: 
Bitping is in fact the least efficient of the apps but it is also the least demanding in terms of traffic. it's still a real pain to have to configure it but as you can see I found a method easy enough for end users. A few cents is better than nothing, surely if in the future the app package becomes too big or heavy it would be one of those that I could eliminate in favor of others.

I would like to hear your opinion (and that of other users) again on these matters

for now I wish you happy holidays 🙂

**Reply by AntonGreydon** [reply]
You are correct on everything you pointed out regarding the Mystnodes. I do not believe however, that their Dvpn method is much different than Honeygain or for that matter all the other bandwith VPN /DVPN services. Packetstream being the least safe one IMO. 

The small MYST fee to transfer in order to open a node does make it cumbersome. So I think you are right and Myst nodes should probably be excluded from your money4band package after all.

**Comment by MRColorR** [comment]
Myst is an open to all dVPN (unless you set it to business only), it may have shadier private clients among the good ones. I agree that packetstream is the closest crypto-free mainstream alternative (they are proxies rather than vpn tunnels though).
Honeygain should grant access only to trustworthy companies (business contracts) by default, unless you're thinking about Honeygain CD but it's not available in linux containers and it's an opt-in option on others platforms.  Do you have more information? ii so i could add them to the FAQ.

In any case, to add an extra protection in addition to the isolation offered by containers, which is already a big step forward compared to apps installed directly on the system, I'm thinking of adding an option to automatically use  DNS in project's containers with additional security features such as 1.1.1.2 from cloudflare or 76.76.2.1 from controlD.

if you have other requests, tips or apps to add let me know, they are always welcome :)

**Comment by johnkennyuk** [comment]
didn't see this post till after i posted a reply to a related post on the issues.

https://github.com/MRColorR/money4band/issues/61#issuecomment-1703823937

__

> I run mysterium node on a RPi and yes it can be bandwidth hog if you get the sessions to use it all, but anyway i came across this app as I want to increase my passive earnings, If I could combine all my earning apps into the one host on docker I may switch node hosts myself, if its added into the compose yml then as with the current apps users can comment out apps they don't want. I can say from experience if you can leave mysterium nodes running 24/7 everyday then eventually you can rack up some high earnings like mine is, it certainly seems its going to provide a higher rate of earnings compared to the other apps already in this app, as i say as long as you keep it running you will earn fair bob especially at the moment where the VPN client payment model has switched to subscription models from pay as you go I'm earning double nearly triple more than with the pay as you go model..
> 
> Anyway, back to the point i do think it could be a popular addition then let end users decide if they use it or not once, they know about its potential bandwidth usages._

_

**Comment by MRColorR** [comment]
no problem :) 
also we're working on it

**Reply by MRColorR** [reply]
see #56 
Please download and try the new pre-release to test the newer apps and functions  https://github.com/MRColorR/money4band/releases

**Reply by MRColorR** [reply]
all these changes have been pushed into main and are now part of the latest version.
enjoy
